sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2AlterTableCommands.scala

```diff
@@ -208,13 +209,25 @@ case class AlterColumnSpec(
     newNullability: Option[Boolean],
     newComment: Option[String],
     newPosition: Option[FieldPosition],
-    newDefaultExpression: Option[String]) extends Expression with Unevaluable {
+    newDefaultExpression: Option[DefaultValueExpression]) extends Expression with Unevaluable {
 
-  override def children: Seq[Expression] = Seq(column) ++ newPosition.toSeq
+  override def children: Seq[Expression] = Seq(column) ++ newPosition.toSeq ++
+    newDefaultExpression.toSeq
   override def nullable: Boolean = false
   override def dataType: DataType = throw new UnresolvedException("dataType")
-  override protected def withNewChildrenInternal(newChildren: IndexedSeq[Expression]): Expression =
-    copy(column = newChildren(0).asInstanceOf[FieldName])
+  override protected def withNewChildrenInternal(
+      newChildren: IndexedSeq[Expression]): Expression = {
+    val newColumn = newChildren(0).asInstanceOf[FieldName]
+    val newPosition = newChildren collectFirst {
```

Comment:
we can avoid instance checking
```
val newPos = if (newPosition.isDefined) {
  Some(newChildren(1).asInstanceOf[FieldPosition])
} else {
  None
}
val newDefault = if (newDefaultExpression.isDefined) {
  Some(newChildren.last.asInstanceOf[DefaultValueExpression])
} else {
  None
}
```

Reply:
@cloud-fan unfortunately the fieldPosition is not always set, so sometimes position 1 is fieldPosition, other times its DefaultValueExpression.  Any suggestion?

Reply:
nice idea, done